spark/v3.3/spark/src/main/java/org/apache/iceberg/spark/actions/ExpireSnapshotsSparkAction.java

```diff
@@ -139,6 +152,14 @@ public ExpireSnapshotsSparkAction deleteWith(Consumer<String> newDeleteFunc) {
     return this;
   }
 
+  public ExpireSnapshotsSparkAction bulkDeleteWith(Consumer<Iterable<String>> newBulkDeleteFunc) {
+    Preconditions.checkArgument(
+        ops.io() instanceof SupportsBulkOperations,
+        "FileIO %s does not support bulk deletion",
+        table.io().getClass().getName());
```

Comment:
I shall probably move this `table.io()` to `ops.io()` for consistency, but given check take 1+hr to build, I will handle it together with other review feedback